Backend/patients_servers/patients_servers.js

```
@@ -105,18 +106,36 @@ app.get("/users/:userId/patients", async (req, res) => {
     if (!patientIds || patientIds.length === 0) {
       return res.status(400).json({ message: "No patient IDs provided" });
     }
-    const patientsData = await Promise.all(patientIds.map(async (id) => {
-      id = parseInt(id);
-      const patient = await prisma.patient.findUnique({ where: { id } });
-      return {
-        id,
-        firstName: patient.firstName,
-        middleName: patient.middleName,
-        lastName: patient.lastName,
-      };
+    const cleanedPatientIds = patientIds.map(id => {
+      const cleanedId = id.replace(/^"|"$/g, '');
+      return BigInt(cleanedId);
+    }).filter(id => id !== null);
+    try {
+        const patientsData = await fetchPatientsData(cleanedPatientIds);
+        res.status(200).json(patientsData);
+    } catch (error) {
+        res.status(500).json({ message: "Failed to fetch patient data" });
+    }
+});
+
+async function fetchPatientsData(patientIds) {
+    const patients = await Promise.all(patientIds.map(async (id) => {
+        const patient = await prisma.patient.findUnique({
+            where: { id }
+        });
+        if (!patient) {
+            console.error("No patient found for ID:", id);
```

Why is this error logged in the console? Won't this fail silently for the user?